2017/proposals/bansal-utkarsh.md

```diff
@@ -0,0 +1,628 @@
+                print(n)
+        >       assert n % 2 == 0 or nn % 2 == 0
+        E       assert ((3 % 2) == 0 or (9 % 2) == 0)
+
+        experiments.py:37: AssertionError
+        --------------------------------- Captured stdout call ---------------------------------
+        3
+        ```
+
+
+    * Cleanup (`cleanup.py`) This plugin deletes XDR offset files after all testing is done. This will have to
+    be re-written as a pytest plugin.
+    * Knownfailure (`knownfailure.py`) This is a decorator to check tests that are expected to fail. Pytest has
+    something similar to this called `xfail`. The documentation for this is available [here](http://doc.pytest.org/en/latest/skipping.html#mark-a-test-function-as-expected-to-fail) .
+    This needs discussion to finalize if we need to port the existing code or if `xfail` is good enough for our use.
+    * Memory leak (`memleak.py`) This plugin detects memory leaks in tests. There is an open-source (MIT license) plugin, [pytest-leaks](https://github.com/abalkin/pytest-leaks).
+    This also needs discussion to finalize if we need to port the existing code or if pytest-leaks is good enough for our use.
+    * Open files (`open_files.py`) This plugin lists all open files when a test fails or errors. They also are listed at
+    the end of the test suite. This plugin is currently disabled because we were facing issue with capturing stdout. Discussion
+    is needed to decide if we need this plugin anymore. If we do, this will have to be rewritten for pytest as no open source
+    alternatives are available.
+    * To run tests in parallel, there is a open source (MIT license) plugin [pytest-xdist](https://github.com/pytest-dev/pytest-xdist) that is recommended by pytest.
+* Configure TravisCI and code coverage. This will not take much time as most of the original config will run just fine. Only a
+couple of commands will have to be changed.
+
+**Challenges** The main challenge in this part is to not loose any tests in the conversion process. Code coverage data can be used to identify parts where this
+ problem occurs.
+
+
+Part 2 
+------
+This part will focus on the coordinates module and its tests. The coordinates module is a very important part of 
+MDAnalysis and any bugs in it will result in error in further calculations. Currently the coordinates module has not 
+been tested in a consistent fashion. Also all the reader/writers don’t necessarily follow the standard API.
+
+* Modify Reader/Writer API for files in the coordinates module for to follow the standard API.
+* Modify coordinates tests to use base Reader/Writer test classes. This will help to set a baseline that all 
+readers/writers will have to conform to and at the same time avoid code duplication among test classes.
+* Modify code to make use of pytest only features such as assert statements and fixtures.
+
+This part will be split into multiple pull requests, ideally one pull request for each coordinate test file.
```

Comment:
Yes, I agree. This is what I meant when I said the timeline feels a bit tight on the mailing list. I would prefer making this an optional goal instead of trying to do it in parallel with other stuff.